ml-proto/test/func.wast

```diff
@@ -0,0 +1,442 @@
+  (func "local-first-f32" (local f32 f32) (result f32) (get_local 0))
+  (func "local-first-f64" (local f64 f64) (result f64) (get_local 0))
+  (func "local-second-i32" (local i32 i32) (result i32) (get_local 1))
+  (func "local-second-i64" (local i64 i64) (result i64) (get_local 1))
+  (func "local-second-f32" (local f32 f32) (result f32) (get_local 1))
+  (func "local-second-f64" (local f64 f64) (result f64) (get_local 1))
+  (func "local-mixed" (local f32) (local $x i32) (local i64 i32) (local) (local f64 i32) (result f64)
+    (f32.neg (get_local 0))
+    (i32.eqz (get_local 1))
+    (i64.eqz (get_local 2))
+    (i32.eqz (get_local 3))
+    (f64.neg (get_local 4))
+    (i32.eqz (get_local 5))
+    (get_local 4)
+  )
+
+  ;; Typing of parameters
+
+  (func "param-first-i32" (param i32 i32) (result i32) (get_local 0))
+  (func "param-first-i64" (param i64 i64) (result i64) (get_local 0))
+  (func "param-first-f32" (param f32 f32) (result f32) (get_local 0))
+  (func "param-first-f64" (param f64 f64) (result f64) (get_local 0))
+  (func "param-second-i32" (param i32 i32) (result i32) (get_local 1))
+  (func "param-second-i64" (param i64 i64) (result i64) (get_local 1))
+  (func "param-second-f32" (param f32 f32) (result f32) (get_local 1))
+  (func "param-second-f64" (param f64 f64) (result f64) (get_local 1))
+  (func "param-mixed" (param f32 i32) (param) (param $x i64) (param i32 f64 i32) (result f64)
+    (f32.neg (get_local 0))
+    (i32.eqz (get_local 1))
+    (i64.eqz (get_local 2))
+    (i32.eqz (get_local 3))
+    (f64.neg (get_local 4))
+    (i32.eqz (get_local 5))
+    (get_local 4)
+  )
+
+  ;; Typing of result
+
+  (func "empty")
+  (func "value-void" (call $dummy))
+  (func "value-drop" (i32.const 1))
+  (func "value-i32" (result i32) (i32.const 77))
+  (func "value-i64" (result i64) (i64.const 7777))
+  (func "value-f32" (result f32) (f32.const 77.7))
+  (func "value-f64" (result f64) (f64.const 77.77))
+  (func "value-block-void" (block (i32.const 1) (call $dummy)))
+  (func "value-block-drop" (block (call $dummy) (i32.const 1)))
+  (func "value-block-i32" (result i32) (block (call $dummy) (i32.const 77)))
+
+  (func "return-nullary" (return))
+  (func "return-void" (return (call $dummy)))
+  (func "return-drop" (return (i32.const 1)))
+  (func "return-i32" (result i32) (return (i32.const 78)))
+  (func "return-i64" (result i64) (return (i64.const 7878)))
+  (func "return-f32" (result f32) (return (f32.const 78.7)))
+  (func "return-f64" (result f64) (return (f64.const 78.78)))
+  (func "return-block-void" (return (block (i32.const 1) (call $dummy))))
+  (func "return-block-drop" (return (block (call $dummy) (i32.const 1))))
+  (func "return-block-i32" (result i32)
+    (return (block (call $dummy) (i32.const 77)))
+  )
+
+  (func "break-nullary" (br 0))
+  (func "break-void" (br 0 (call $dummy)))
+  (func "break-drop" (br 0 (i32.const 1)))
+  (func "break-i32" (result i32) (br 0 (i32.const 79)))
```

Comment:
Yes. In particular, the spec defines `return` as a shorthand for the corresponding break.